network.go

```diff
@@ -62,6 +62,12 @@ type NetworkController interface {
 
 	// WalkNetworks uses the provided function to walk the Network(s) managed by this controller.
 	WalkNetworks(walker NetworkWalker)
+
+	// NetworkByName returns the Network which has the passed name, if it exists otherwise nil is returned
+	NetworkByName(name string) Network
```

Comment:
isnt it better to return a pointer to Network and hence it can return a nil if the query fails ?
The same comment applies to all other query methods.